runtime/runtime-params-estimator/src/lib.rs

```diff
@@ -1042,9 +1045,115 @@ fn storage_remove_ret_value_byte(ctx: &mut EstimatorContext) -> GasCost {
 }
 
 fn touching_trie_node(ctx: &mut EstimatorContext) -> GasCost {
-    // TODO: Actually compute it once our storage is complete.
-    // TODO: temporary value, as suggested by @nearmax, divisor is log_16(20000) ~ 3.57 ~ 7/2.
-    storage_read_base(ctx) * 2 / 7
+    let read = touching_trie_node_read(ctx);
+    let write = touching_trie_node_write(ctx);
+    return std::cmp::max(read, write);
+}
+
+fn touching_trie_node_read(ctx: &mut EstimatorContext) -> GasCost {
+    if let Some(cost) = ctx.cached.touching_trie_node_read.clone() {
+        return cost;
+    }
+    let warmup_iters = ctx.config.warmup_iters_per_block;
+    let measured_iters = ctx.config.iter_per_block;
+    let mut testbed = ctx.testbed();
+    let tb = testbed.transaction_builder();
+
+    // Number of bytes in the final key. Will create 2x that many nodes.
+    // Picked somewhat arbitrarily, balancing estimation time vs accuracy.
+    let final_key_len = 1000;
+
+    // Prepare a long chain in the trie
+    let signer = tb.random_account();
+    let key = "j".repeat(final_key_len);
+    let mut setup_block = Vec::new();
+    for key_len in 0..final_key_len {
+        setup_block.push(tb.account_insert_key(signer.clone(), &key.as_str()[..key_len], "0"));
```

Comment:
```suggestion
        let key = &key[..key_len];
        let value = "0";
        setup_block.push(tb.account_insert_key(signer.clone(), key, value));
```

mostly to make `"0"` more obvious